### PR TITLE
vengeance ability

### PR DESCRIPTION
**Pull request by ezrn**

### CODE REVIEW

File: champions/src/main/java/me/mykindos/betterpvp/champions/champions/skills/skills/knight/passives/Vengeance.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.EventPriority;
+import org.bukkit.event.Listener;
+import org.bukkit.event.entity.EntityDamageEvent.DamageCause;
+
+@Singleton
+@BPvPListener
+public class Vengeance extends Skill implements PassiveSkill, Listener {
+
+    private int numHits;
```
Comment (StudentAlleg) [inline champions/src/main/java/me/mykindos/betterpvp/champions/champions/skills/skills/knight/passives/Vengeance.java:24]: Num hits is never retrieved. Needs to override loadSkillConfig: `@Override
    public void loadSkillConfig(){
        numHits= getConfig("numHits", 3, Integer.class);
    }`

File: champions/src/main/java/me/mykindos/betterpvp/champions/champions/skills/skills/knight/passives/Vengeance.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    }
+
+    @Override
+    public String getName() {
+        return "Vengeance";
+    }
+
+    @Override
+    public String[] getDescription(int level) {
+        return new String[]{
```
Comment (StudentAlleg) [inline champions/src/main/java/me/mykindos/betterpvp/champions/champions/skills/skills/knight/passives/Vengeance.java:38]: No usage of numHits in desciption.

**Review by StudentAlleg — CHANGES_REQUESTED** [review]
numHits is not implemented correctly. Is this supposed to be the maximum number of hits, a tracker of hits (in which case do not retrieve it from config), or global hits (in which case it is implemented correctly)?

There does not appear to be a cap for numHits.

File: champions/src/main/java/me/mykindos/betterpvp/champions/champions/skills/skills/knight/passives/Vengeance.java (hunk 1 above)
Comment (StudentAlleg) [reply champions/src/main/java/me/mykindos/betterpvp/champions/champions/skills/skills/knight/passives/Vengeance.java:24]: Does not apply if this is not the maximum hits this skill applies for.

File: champions/src/main/java/me/mykindos/betterpvp/champions/champions/skills/skills/knight/passives/Vengeance.java (hunk 2 above)
Comment (StudentAlleg) [reply champions/src/main/java/me/mykindos/betterpvp/champions/champions/skills/skills/knight/passives/Vengeance.java:38]: Does not apply if this is not the maximum hits this skill applies for.

File: champions/src/main/java/me/mykindos/betterpvp/champions/champions/skills/skills/knight/passives/Vengeance.java [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+                "For every subsequent hit, your damage will increase by <val>" + (level * 0.25) +"</val>",
```
Comment (StudentAlleg) [inline champions/src/main/java/me/mykindos/betterpvp/champions/champions/skills/skills/knight/passives/Vengeance.java:41]: Should probably have damage increase in config.

**Review by StudentAlleg — COMMENTED** [review]
Skill increases damage per hit per player, resseting if damage is taken from any source.

**Review by StudentAlleg — APPROVED** [review]

**Comment by Mykindos** [comment]
too similar to repeated strikes

**Pull request closed without merging**